Design Basic "Error 404" Page

FusionStreak opened the pull request:
Fixes #22

- Created `pageNotFound` component
- Created `assets` directory
- Created `assets/images` directory
- Added `Star_Wars_BB8.svg` temporary image for 404 Page
- Added `file-loader` package for assest management
- Added `package-lock.json` to `.gitignore`, this is generated by `npm install` using `package.json`
- Added `react-router-dom` to `package.json`

## TODO

- CSS need to clean up `PageNotFound`

OussamaSaoudi commented on src/components/PageNotFound.js:
```diff
@@ -0,0 +1,35 @@
+import Container from 'react-bootstrap/Container'
+import Row from 'react-bootstrap/Row'
+import Col from 'react-bootstrap/Col'
+//import { Link, Router } from 'react-router-dom'
```
Modify the comment into a form with 
/* TODO: explain why this is commented out
 * import { Link, ....
 */

OussamaSaoudi commented:
Finally merged router stuff. Feel free to rebase and implement router functionality.

FusionStreak commented:
- Added the `PageNotFound` to the main Router component. It uses a "wildcard" to match/redirect any undefined routes. 
- Switched to using `Link` inside the  `PageNotFound` component to have a smoother redirect to the home page.
- A "Home" route will need to be defined, as `/` is directing to 404 Page right now.
